src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const dataOfState = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        dataOfState.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete stateCopy[key];
+        });
+        dataOfState.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => {
+          delete stateCopy[key];
+        });
+        dataOfState.push({ ...stateCopy });
+        break;
+
+      default:
+        return 'incorect data';
```

Comment:
I think the error message should be more informative, pointing specifically to the action.type